### DISCUSSION TITLE
Report | Week 15

### DISCUSSION THREAD

**Discussion by tech-n1c** [SE-TINF22B6/time2shine#86] [Show and tell]
Hey there 👋

This week, we've been quite busy with other projects, so sadly there wasn't too much progress on time2shine. Still, we have been working on testing different parts of our architecture.

This includes the game engine - PixiJS - where small Unit tests can be done via JavaScript, but for more extensive testing, external frameworks are required. We don't have any experience with JavaScript testing, really, so we spent our time digging into the topic. The same goes for our React website, where we intend to do some integration testing together with the backend. We want to make sure that everything works as intended, from submitting game scores from the games to our database until users check the leaderboards where the data is fetched again from our backend.

The backend itself is being tested via classic Java JUnit tests, and together with the website we want to use new testing frameworks which is - again - something new for us, so we are working on that.

**Comment by sfrancisw** [comment]
On the other hand, the snake game is looking more alive now as we implemented the graphics for the game. It is almost finished now and there are just a few features missing so we're excited about that! We also got rid of a lot of bugs so the snake splits properly and detects a roehrig that is colliding with one of the snakes. Depending on the part that got hit, the snake either splits in two or gets deleted.
![grafik](https://github.com/SE-TINF22B6/time2shine/assets/49996344/0ebddd7d-7448-44cc-814e-986d9f193387)

**Comment by nuo-li** [comment]
For frontend testing, you could have a look into Jest
